docs/src/main/docs/guides/11_config_se_guide.adoc

```diff
@@ -0,0 +1,899 @@
+----
+import static io.helidon.config.ConfigSources.file;
+...
+
+  private static Config buildConfig() {
+      return Config.builder()
+          .sources(
+              file("config-file.properties"), // <1>
+              classpath("application.yaml"))
+          .build();
+  }
+----
+<1> Add a mandatory configuration file.
+
+[source,bash]
+.Build and run the application, then invoke the endpoint and check the response:
+----
+curl http://localhost:8080/greet
+...
+{
+  "message": "HelloFrom-config-file.properties World!"
+}
+----
+<1> The configuration property from the file `config-file.properties` takes precedence.
+
+NOTE: If you want the configuration file to be optional, you must use the `optional` method with `sources`,
+otherwise Helidon will generate an error during startup as shown below.  This is true for both `file` and
+`classpath` sources.  By default, these sources are mandatory.
+
+[source,java]
+.Update the `Main` class and replace the `buildConfig` method:
+----
+  private static Config buildConfig() {
+      return Config.builder()
+          .sources(
+              file("missing-file"), // <1>
+              classpath("application.yaml"))
+          .build();
+  }
+----
+<1> Specify a file that doesn't exist.
+
+[source,bash]
+.Build then start the application and you will see the following output:
+----
+Exception in thread "main" io.helidon.config.ConfigException: Cannot load data from mandatory source FileConfig[missing-file]. File `missing-file` not found.
+----
+
+To fix this, use the `optional` method as shown below, then rerun the test.
+
+----
+...
+    file("missing-file").optional(), // <1>
+
+----
+<1> The `missing-file` configuration file is now optional.
+
+=== Directory source
+
+A directory source treats every file in the directory as a key, and the file contents as the value.  The
+following example includes a directory source as highest precedence.
+
+[source,bash]
+.Create a new directory `helidon-quickstart-se/conf` then create a file named `app.greeting` in that directory with the following contents:
+----
+HelloFromFileInDirectoryConf
+----
+
+[source,java]
+.Update the `Main` class; 1) Add new import and 2) Replace the `buildConfig` method:
+----
+import static io.helidon.config.ConfigSources.directory;
+...
+
+  private static Config buildConfig() {
+      return Config.builder()
+          .sources(
+              directory("conf"), // <1>
+              classpath("config.properties").optional(),
+              classpath("application.yaml"))
+          .build();
+  }
+----
+<1> Add a mandatory configuration directory.
+
+[source,bash]
+.Build and run the application, then invoke the endpoint and check the response:
+----
+curl http://localhost:8080/greet
+...
+{
+  "message": "HelloFromFileInDirectoryConf World!"
+}
+----
+<1> The greeting was fetched from the file named `app.greeting`.
+
+==== Exceeding three sources
```

Comment:
@tomas-langer Why don't we add a varargs variant:`Config.sources(Supplier<ConfigSource>... sources)` so that this section can be removed? Given that we do have the `ConfigSources.create()` method (with `@SafeVarArgs`), I see no reason not to.

Reply:
@tomas-langer Oh, wait, I hadn't noticed that it was a default method; never mind.